src/common/datasource/src/object_store.rs

```diff
@@ -44,17 +54,47 @@ pub fn parse_url(url: &str) -> Result<(String, Option<String>, String)> {
 }
 
 pub fn build_backend(url: &str, connection: &HashMap<String, String>) -> Result<ObjectStore> {
-    let (schema, host, _path) = parse_url(url)?;
+    let (schema, host, path) = parse_url(url)?;
+    let (root, _) = find_dir_and_filename(&path);
 
     match schema.to_uppercase().as_str() {
```

Comment:
So we don't support azblob, aliyun oss etc. in v0.4? @waynexia @fengjiachun

Reply:
Looks like it's only used in file table.

Reply:
I'm not sure. But it won't be too much work to add support for other backends that opendal supported